cmd/osbuild-worker/main.go

```diff
@@ -84,6 +88,68 @@ func WatchJob(ctx context.Context, job worker.Job) {
 	}
 }
 
+// checks if the instance is running on EC2, it has some potential for false positives.
+func isRunningOnAWS() bool {
+	// first check for older instances
+	bUuid, err := os.ReadFile("/sys/hypervisor/uuid")
+	if err == nil {
+		return strings.HasPrefix(string(bUuid), "ec2")
+	}
+	// second check for m5/c5 instances, works because the worker runs as root
+	bUuid, err = os.ReadFile("/sys/devices/virtual/dmi/id/product_uuid")
+	if err == nil {
+		return strings.HasPrefix(string(bUuid), "EC2")
+	}
+	return false
+}
+
+// protect an AWS instance from scaling and/or terminating.
+func setProtection(protected bool) error {
+	// create a new session
+	awsSession, err := session.NewSession()
+	if err != nil {
+		logrus.Error("Error getting AWS session")
+		return err
+	}
+
+	// get the identity for the instanceID
+	identity, err := ec2metadata.New(awsSession).GetInstanceIdentityDocument()
+	if err != nil {
+		logrus.Error("Error getting identity")
```

Comment:
Shouldn't be an error imo, the instance identity document is one of the main ways we can identify if we're running on aws.